pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -609,6 +609,18 @@ func getVirtualHostsForSniffedServicePort(vhosts []*route.VirtualHost, routeName
 	return virtualHosts
 }
 
+func IgnorePortInHostMatching(routeName string, listenerPort int, node *model.Proxy) bool {
+	// When generating RDS for ports created via the SidecarScope, we treat ports as HTTP proxy style ports
+	// if ports protocol is HTTP_PROXY.
+	egressListener := node.SidecarScope.GetEgressListenerForRDS(listenerPort, routeName)
+	if egressListener != nil && egressListener.IstioListener != nil && egressListener.IstioListener.Port != nil &&
+		protocol.Parse(egressListener.IstioListener.Port.Protocol) == protocol.HTTP_PROXY {
+		listenerPort = 0
+	}
```

Comment:
This seems inappropriate here.. 

Don't we already do it in BuildSidecarOutboundVirtualHosts?

We shouldn't have two places we do this. its expensive and makes the logic hard to follow

Reply:
yes, it is indeed repetitive code. I will fix it.